pkg/service/repair/controller.go

```diff
@@ -177,50 +172,39 @@ func (c *rowLevelRepairController) TryBlock(hosts []string) (bool, allowance) {
 	return true, a
 }
 
+func (c *rowLevelRepairController) parallelLimit() int {
+	p := c.intensity.Parallel()
+
+	if p <= defaultParallel {
+		return c.intensity.MaxParallel()
+	}
+
+	return p
+}
+
 func (c *rowLevelRepairController) shouldBlock(hosts []string, intensity float64) bool {
 	// ALLOW if nothing is running
 	if !c.Busy() {
 		return true
 	}
 
-	// DENY if any host has max nr of jobs already running
+	// DENY if any host has a job running already. We don't allow more than a single repair job per host.
 	for _, h := range hosts {
-		if c.jobs[h] >= c.limits[h].Default {
+		if c.jobs[h] >= 1 {
```

Comment:
I am curious what was c.limits[h].Default?

Reply:
AFAIR it's initialized in this hunk: https://github.com/scylladb/scylla-manager/pull/3506#discussion_r1289704818

So, technically, before this PR we allowed up to "number of shards" concurrent repair jobs in a single replica (other conditions aside).

Reply:
AFAIR it's initialized in this hunk: https://github.com/scylladb/scylla-manager/pull/3506#discussion_r1289704818

So, technically, before this PR we allowed up to "number of shards" concurrent repair jobs in a single replica (other conditions aside).

This was the central part of the "bug" - SM code was mixing the "repair ranges math" with the "repair jobs" math.

It most probably came from the original repair code that was per-shard-ranges based. And probably it made sense then. But for a row-level repair things can and should be significantly simplified.

Reply:
Yes, this is the most important change in this PR.